src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationSender.java

```diff
@@ -49,28 +57,43 @@ public void sendReconciliationReport(
     ) throws IOException, SendEmailException {
         Map<String, File> attachments = new HashMap<>();
 
-        File file = reconciliationCsvWriter.writeSummaryReconciliationToCsv(summaryReport);
-        attachments.put(
-            getReportAttachmentName(ATTACHMENT_SUMMARY_PREFIX, date),
-            file
-        );
-
-        if (detailedReport != null) {
-            File detailedReportFile = reconciliationCsvWriter
-                .writeDetailedReconciliationToCsv(detailedReport);
+        if (willReportsBeAttached(summaryReport, detailedReport)) {
```

Comment:
i think they just do not want to get empty file. if there is no discrepancy we do not need to add file?

Reply:
I decided they do not want to receive emails with empty report - emails is ket word. Will check with Sanchta.